### PR TITLE
Enable multi instrumentation by default

### LINKED ISSUE #3090: Enable "enable-multi-instrumentation" flag by default
### Component(s)

collector, auto-instrumentation

### Is your feature request related to a problem? Please describe.

The features that are enabled by `enable-multi-instrumentation` have been there for a while and it might be a good time to enable them by default.

The only risk I can see is those user workloads using the  `instrumentation.opentelemetry.io/container-names` would not be instrumented when they are restarted or recreated. We can implement a warning as part of the validation webhook + something as part of the upgrade routines in order to replace this annotation with the language-specific ones.

### Describe the solution you'd like

.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by iblancasa**
Closes #3090

### CODE REVIEW

**Review by IshwarKanse — APPROVED** [review]

File: .chloggen/3090-enable-multiinstrumentation-by-default.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+# A brief description of the change. Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Enable multi instrumentation by default.
+
+# One or more tracking issues related to the change
+issues: [3090]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
```
Comment (swiatekm) [inline .chloggen/3090-enable-multiinstrumentation-by-default.yaml:16]: We should elaborate how this change can break things for users here.

**Review by swiatekm — APPROVED** [review]

File: .chloggen/3090-enable-multiinstrumentation-by-default.yaml (hunk 1 above)
Comment (jaronoff97) [reply .chloggen/3090-enable-multiinstrumentation-by-default.yaml:16]: +1

**Comment by jaronoff97** [comment]
should we add some warnings as we discussed [here](https://github.com/open-telemetry/opentelemetry-operator/issues/3090#issuecomment-2206875946)?

**Comment by iblancasa** [comment]
> should we add some warnings as we discussed [here](https://github.com/open-telemetry/opentelemetry-operator/issues/3090#issuecomment-2206875946)?

I don't think so since #3213 was implemented. With this, the warnings are not needed because users will not need to do anything.

File: .chloggen/3090-enable-multiinstrumentation-by-default.yaml (hunk 1 above)
Comment (iblancasa) [reply .chloggen/3090-enable-multiinstrumentation-by-default.yaml]: I think now is much better.

**Comment by iblancasa** [comment]
Can we merge this?

**Comment by swiatekm** [comment]
@jaronoff97 @pavolloffay can you have a look as well? I don't want to merge a change like this without at least another maintainer approval.

**Review by jaronoff97 — APPROVED** [review]
I think this LGTM. i don't believe there's anything here where an existing user would be functionally broken though right? Just more possible error states where users try to do both methods?

**Comment by jaronoff97** [comment]
also, looks like tests are failing

**Comment by iblancasa** [comment]
> I think this LGTM. i don't believe there's anything here where an existing user would be functionally broken though right?

I don't think so. 

> Just more possible error states where users try to do both methods?

Yes. And that should not be happening right now so..

> also, looks like tests are failing

Yes. It is related to #3307.

**Pull request merged**